Graph/Traversal/BreadthFirstSearch.h

```diff
@@ -0,0 +1,80 @@
+#ifndef BREATHFIRSTSEARCH_H
+#define BREATHFIRSTSEARCH_H
+
+#include <cstdlib>
+#include <vector>
+#include <list>
+#include <iostream>
+
+class Graph {
+    /* Graph class representing an undirected weighted graph */
+        int num_vertices;
+        std::vector< std::list< std::pair<int,int> > > adj_list;
+        /* adjacency list representing graph
+         * first int represents destination node
+         * second int represents weight of the edge
+         */
+    public:
+        Graph(int num_vertices) {
+            this->num_vertices = num_vertices;
+            adj_list.resize(num_vertices);
+        }
+        void addUndirectedEdge(int source_vertex, int dest_vertex, int weight) {
+            std::pair<int, int> vertexPair = std::make_pair(dest_vertex, weight);
+            adj_list[source_vertex].push_back(vertexPair);
+            /* undirected graph so we add the edge both ways */
+            vertexPair = std::make_pair(source_vertex, weight);
+            adj_list[dest_vertex].push_back(vertexPair);
+        }
+        bool isValidSource(int source_vertex) { return !(source_vertex < 0 || source_vertex >= num_vertices); }
+        /* prints BFS-traversal from a source vertex */
+        void breadthFirstSearch(int source_vertex);
+        /* prints DFS-traversal from a source vertex */
+        void depthFirstSearch(int source_vertex) { /* TO-DO */ };
+};
+
+Graph *generateGraph() {
```

Comment:
Make seperate functions for getting inputs from the user, such as:
- number of vertices
- number of edges
- edge info (source, destination and weight)

and make this function call those.

Also, since all these functions (including this one) are essential for a graph to have, make these member functions of the `Graph` class.